keras_cv/models/densenet.py

```diff
@@ -0,0 +1,296 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""DenseNet models for KerasCV.
+
+Reference:
+  - [Densely Connected Convolutional Networks](https://arxiv.org/abs/1608.06993)
+"""
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import backend
+from tensorflow.keras import layers
+
+BN_AXIS = 3
+
+
+def dense_block(x, blocks, name):
+    """A dense block.
+
+    Args:
+      x: input tensor.
+      blocks: integer, the number of building blocks.
+      name: string, block label.
+
+    Returns:
+      Output tensor for the block.
+    """
+    for i in range(blocks):
+        x = conv_block(x, 32, name=name + "_block" + str(i + 1))
+    return x
+
+
+def transition_block(x, reduction, name):
```

Comment:
What do you think about this kind of intermediate block/component?

Are these general enough to be exposed to build/contribute other networks?

Reply:
For now we wouldn't be exposing any modeling related component as public API.

Reply:
+1, let's not expose such sub-model APIs.

Reply:
I do not think we should expose these either - at least not for the time being.